heat/core/tests/test_signal.py

```diff
@@ -119,3 +120,29 @@ def test_convolve(self):
 
         conv = ht.convolve(1, 5)
         self.assertTrue(ht.equal(ht.array([5]), conv))
+
+        # test batched convolutions, distributed along the first axis
+        signal = ht.random.randn(1000, dtype=ht.float64)
+        batch_signal = ht.empty((10, 1000), dtype=ht.float64, split=0)
+        batch_signal.larray[:] = signal.larray
+        kernel = ht.random.randn(19, dtype=ht.float64)
+        batch_convolved = ht.convolve(batch_signal, kernel, mode="same")
+        self.assertTrue(ht.equal(ht.convolve(signal, kernel, mode="same"), batch_convolved[0]))
```

Comment:
Just to be sure, I would test a couple more of the batches against the original signal, at least one per rank.

Reply:
added a check on the last rank. The signal is repeated across the -1 dimension, so the result really should be the same everywhere.